terraform/modules/cds_export/kms.tf

```diff
@@ -0,0 +1,50 @@
+resource "aws_kms_key" "this" {
+  description         = "KMS key used to encrypt the CDS export bucket."
+  enable_key_rotation = true
+}
+
+resource "aws_kms_alias" "this" {
+  name          = "alias/${aws_s3_bucket.export.id}"
+  target_key_id = aws_kms_key.this.key_id
+}
+
+resource "aws_kms_key_policy" "this" {
+  key_id = aws_kms_key.this.id
+  policy = data.aws_iam_policy_document.kms.json
+}
+
+data "aws_iam_policy_document" "kms" {
+  #checkov:skip=CKV_AWS_109:Only assigning the kms:GenerateDataKey and kms:Decrypt permissions led to a 'you won't be able to manage the key once created' error
+  #checkov:skip=CKV_AWS_111:Only assigning the kms:GenerateDataKey and kms:Decrypt permissions led to a 'you won't be able to manage the key once created' error
+  #checkov:skip=CKV_AWS_356:Only assigning the kms:GenerateDataKey and kms:Decrypt permissions led to a 'you won't be able to manage the key once created' error
+  statement {
+    principals {
+      type        = "AWS"
+      identifiers = [var.ecs_task_role_arn]
```

Comment:
Here, won't the task role only need to be able to encrypt, while the cds role will need to be able to decrypt?

Reply:
Yeah I think I agree, the CDS role is also optionally created so this will need to be a separate policy document, that is only applied if the role exists. Will update 👍

Reply:
Done 👍

Reply:
I had some issues getting the CDS role policy _and_ the ECS task role to combine and apply to the key (when required), but this has now been fixed by this [commit](https://github.com/communitiesuk/submit-social-housing-lettings-and-sales-data-infrastructure/pull/80/commits/1fade63b3daecc65b646f600754a7dd70ddad946). I don't love the implementation but it's the only solution I've managed to get working on the key!